src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,792 @@
+                            return true;
+                        }
+                    }
+                }
+
+                documentUpdateInfo = default;
+                return false;
+            }
+
+            private bool TryGetTypeSyntax(ExpressionSyntax expression, out TypeSyntax typeSyntax)
+            {
+                var typeSymbol = _semanticModel.GetTypeInfo(expression, _cancellationToken).Type;
+                if (typeSymbol.TypeKind == TypeKind.Error ||
+                    typeSymbol.ContainsAnonymousType())
+                {
+                    typeSyntax = default;
+                    return false;
+                }
+
+                typeSyntax = typeSymbol.GenerateTypeSyntax();
+                return true;
+            }
+
+            private string GetFreeSymbolName(string prefix, int position)
+                => NameGenerator.GenerateUniqueName(
+                    prefix,
+                    _semanticModel.LookupSymbols(position).Select(symbol => symbol.Name).ToImmutableHashSet(),
+                    StringComparer.CurrentCulture);
+
+            private bool TryConvertIfInForEach(QueryExpressionSyntax source, ForEachStatementSyntax forEachStatement, out DocumentUpdateInfo documentUpdateInfo)
+            {
+                // before foreach(var x in from a in b select a)
+                var statement = forEachStatement.Statement;
+                StatementSyntax internalNodeMethod(ExpressionSyntax expression)
+                {
+                    // before 
+                    //  foreach (var a in from a in b where a > 5 select a) 
+                    //  { 
+                    //      dosomething(a); 
+                    //  }
+                    // after 
+                    //  foreach (var a in b)
+                    //  {
+                    //      if (a > 5)
+                    //      {
+                    //          dosomething(a); 
+                    //      }
+                    //  }
+                    if (expression is IdentifierNameSyntax identifier &&
+                        identifier.Identifier.ValueText == forEachStatement.Identifier.ValueText)
+                    {
+                        return statement.WithAdditionalAnnotations(Formatter.Annotation);
+                    }
+                    else
+                    {
+                        // before foreach(var x in from ... a) { dosomething(x); }
+                        // after 
+                        // foreach (var a in ...)
+                        // ...
+                        // {
+                        //      var x = a;
+                        //      dosomething(x); 
+                        //  }
+                        var declaration = SyntaxFactory.LocalDeclarationStatement(
+                            SyntaxFactory.VariableDeclaration(
+                                VarNameIdentifier,
```

Comment:
>VarNameIdentifier [](start = 32, length = 17)

Why do we assume that the declaration was using ```var```? I think we should use type from the original declaration